backend/de.metas.cucumber/src/test/java/de/metas/cucumber/stepdefs/shipment/M_InOut_StepDef.java

```diff
@@ -355,6 +363,20 @@ public void locate_shipment_by_scheduleId(@NonNull final DataTable table)
 		}
 	}
 
+	@And("^validate no M_InOut found for C_Order identified by (.*)$")
+	public void no_M_InOut_found(@NonNull final String orderIdentifier)
+	{
+		final I_C_Order order = orderTable.get(orderIdentifier);
+		assertThat(order).isNotNull();
+
+		final I_M_InOut inOut = queryBL.createQueryBuilder(I_M_InOut.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(COLUMNNAME_C_Order_ID, order.getC_Order_ID())
+				.create()
+				.firstOnlyOrNull(I_M_InOut.class);
```

Comment:
🤔 AFAIU if there are two M_InOuts, then this step will succeed although it should fail

Reply:
🤯 right -> I've looked at it as a firstOnlyOptional